### ISSUE TITLE
pgzero: multiple "Variable" is not defined warnings

### ISSUE THREAD

**Issue by fguillen** [microsoft/pyright#1688]
**Describe the bug**
I am developing using [pgzero library](https://pygame-zero.readthedocs.io/en/stable/) and I see that Pylance is complaning about many of the definitions to not be found

This is specially annoying since I am introducing kids to programming and they are very incomfortable with all these warnings around (which is good), teaching them to ignore warnings is not a good thing I think :(

**To Reproduce**
This is a totally valid pgzero code:

```
import pgzrun

WIDTH = 480
HEIGHT = 340
BACKGROUND_COLOR = (36, 34, 31)

player = Actor("arrow")

def draw():
    screen.fill(BACKGROUND_COLOR)
    player.draw()

def log():
    print("log")

clock.schedule_unique(log, 1)

pgzrun.go()
```

**Expected behavior**
It should be not any "not defined warnings" but there are 3:

> "Actor" is not defined Pylance
> "screen" is not defined Pylance
> "clock" is not defined Pylance

I can hide the warning in  `Actor` and `clock` by adding:

    from pgzero.builtins import Actor
    from pgzero.builtins import clock

but for `screen` I didn't find any solution :/

**Screenshots or Code**

- https://pasteboard.co/JUgQX0b.jpg

**VS Code extension or command-line**

I am running pyright by pylance as an extension in VS Code

- Pyrigh: v1.1.125
- Pylance: v2021.3.3

**Comment by erictraut** [comment]
Thanks for your efforts to develop this library and introduce kids to programming. That's very inspiring!

How does your library normally work without the `import` statements? Are you injecting these symbols into the `builtins` namespace? If so, I'd _highly_ encourage you _not_ to do that. While a dynamic language like Python makes this possible, it's a very bad idea. Not only does it pollute the global namespace, but static type checking tools cannot reason about it. Instead, you should export the public symbols for the library through the primary module `pgzero`, and consumers of the library should access them using:

```python
import pgzero

new pgzero.Actor()
```

or

```python
from pgzero import Actor

Actor()
```

**Comment by erictraut** [comment]
Digging into your library code more, I see that you are injecting these symbols into the namespace of the `__main__` module. My comments above apply here too. It's a very bad idea to inject attributes into namespaces that your library does not own. I'm guessing that your intent is to hide some of the complexity, but I encourage you to reconsider this approach.

In any case, I don't think there's anything we can do to eliminate the errors in Pylance since you're doing something that cannot be supported by static type checking.

**Comment by erictraut** [comment]
A few options for you...

First, you can explicitly import the symbols as I showed above. This is the best approach because it gives you all the benefits of type checking.

Second, if you are not interested in type checking, you can disable all of pyright's type checking features by setting `python.analysis.typeCheckingMode` to "off".

**Comment by yiqunc** [comment]
add this line after the imports will stop Pylance complaining.
screen: pgzero.screen.Screen

the true problem is Pylance still cannot sense any method defined in the screen object.....
